challenge_7/scala/ndsmith3/MissingNum.scala

```diff
@@ -0,0 +1,15 @@
+import scala.annotation.tailrec
+
+object MissingNum {
+  def missingNum(nums: Array[Int]): Int = {
+    @tailrec
+    def test(x: Int): Int = if (!nums.contains(x)) x else test(x + 1)
```

Comment:
This doesn't meet the challenges stated constraint of O(N) time. The recursion is equivalent to a O(N) loop across the array and the contains array lookup is additional overhead beyond O(N).